### PR TITLE
hlb, support: ActiveContractsService

### PR DESCRIPTION
**Pull request by nickchapman-da**
Haskell Ledger Bindings:
- support: ActiveContractsService
- rpc: getActiveContracts

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/hs/bindings/src/DA/Ledger/Convert.hs [hunk 1]
```diff
@@ -229,7 +244,7 @@ raiseTreeEvent = \case
         eid <- raiseEventId exercisedEventEventId
         cid <- raiseContractId exercisedEventContractId
         tid <- perhaps "exercisedEventTemplateId" exercisedEventTemplateId >>= raiseTemplateId
-        ccEid <- undefined exercisedEventContractCreatingEventId
+        ccEid <- raiseEventId exercisedEventContractCreatingEventId
```
Comment (nickchapman-da) [inline language-support/hs/bindings/src/DA/Ledger/Convert.hs]: This is an unrelated bug fix

File: language-support/hs/bindings/src/DA/Ledger/Convert.hs [hunk 2]
```diff
@@ -159,6 +161,19 @@ optional = \case
     Left _ -> Nothing
     Right a -> Just a
 
+raiseGetActiveContractsResponse :: LL.GetActiveContractsResponse -> Perhaps (AbsOffset,Maybe WorkflowId,[Event])
+raiseGetActiveContractsResponse = \case
```
Comment (hurryabit) [inline language-support/hs/bindings/src/DA/Ledger/Convert.hs]: There's no need for a `case` expression here. The pattern can just come after the function name. Record wild cards might also help to make this a bit less verbose.

File: language-support/hs/bindings/test/DA/Ledger/Tests.hs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 -- SPDX-License-Identifier: Apache-2.0
 
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DuplicateRecordFields #-}
```
Comment (hurryabit) [inline language-support/hs/bindings/test/DA/Ledger/Tests.hs]: Are we sure we need this?

File: language-support/hs/bindings/src/DA/Ledger/Convert.hs (hunk 2 above)
Comment (nickchapman-da) [reply language-support/hs/bindings/src/DA/Ledger/Convert.hs]: Are we really happy with the use of `{..}` ?
The way it implicitly brings unknown names into scope seems rather scary.

File: language-support/hs/bindings/test/DA/Ledger/Tests.hs (hunk 3 above)
Comment (nickchapman-da) [reply language-support/hs/bindings/test/DA/Ledger/Tests.hs]: yes, for the `signatories` field label

File: language-support/hs/bindings/src/DA/Ledger/Convert.hs (hunk 2 above)
Comment (hurryabit) [reply language-support/hs/bindings/src/DA/Ledger/Convert.hs]: I'm not the biggest fan of `{..}` either but we have plenty of it in our code base already. Instead of
```
f C{x, y} = x+y
```
or
```
f C{..} = x+y
```
I usually prefer
```
f x = x c + y c
```
That avoids the `{..}` and repeating the names at the same time.
